userspace/libsinsp/parsers.cpp

```diff
@@ -4499,39 +4499,28 @@ void sinsp_parser::parse_setgid_exit(sinsp_evt *evt)
 	}
 }
 
-static std::string convert_to_string(Json::ValueType other) {
-	std::string value_type("Not a valid json type");
-	switch(other) {
-	case Json::ValueType::nullValue:
-		return "nullValue";
-	case Json::ValueType::intValue:
-		return "intValue";
-	case Json::ValueType::uintValue:
-		return "uintValue";
-	case Json::ValueType::realValue:
-		return "realValue";
-	case Json::ValueType::stringValue:
-		return "stringValue";
-	case Json::ValueType::booleanValue:
-		return "booleanValue";
-	case Json::ValueType::arrayValue:
-		return "arrayValue";
-	case Json::ValueType::objectValue:
-		return "objectValue";
-	default:
-		break;
+namespace
+{
+	std::string generate_error_message(const Json::Value& value, const char* field) {
+		std::string val_as_string = value.isConvertibleTo(Json::stringValue) ? value.asString().c_str() : "value not convertible to string";
```

Comment:
for sake of simplicity, i think we can get rid of this logic, no? isn't logging the "field" value sufficient?

Reply:
I disagree. In the context of the segV we know exactly which field was in error. However, we do not know why ? If we have the value - we will have valuable information for debugging.

Reply:
oh yeah, its my fault, you are right to have value logged as well.